src/avx512bw/avx512bw-from-utf8.inl.cpp

```diff
@@ -0,0 +1,462 @@
+/*
+validate_utf8_structure only checks if leading bytes
+are follewed by proper number of continuation bytes
+and then there's another leading byte.
+
+We don't handle end-of-stream here.
+*/
+bool validate_utf8_structure(__m512i input) {
+    /* 1. bitmask for various character byte classes.
+
+        leading: 111010011100011010001
+                 abccdddefgggghiijjjjk   a-k -- 11 UTF8- chars characters
+
+        ascii:   110000011000010000001
+                 ab     ef    h      k
+
+        2 bytes: 001000000000001000000
+                   cc          ii
+
+        3 bytes: 000010000000000000000
+                     ddd              
+
+        4 bytes: 000000000100010010000
+                          gggg   jjjj
+    */
+    uint64_t leading;
+    uint64_t ascii;
+    uint64_t _2bytes;
+    uint64_t _3bytes;
+    uint64_t _4bytes;
+
+    // we can valide 60 - 4 leading bytes
```

Comment:
'valide' is probably polish or something. :-)

Reply:
Sorry, didn't spell check the comments. :(